module/VuFind/src/VuFind/Db/Service/SearchService.php

```diff
@@ -138,17 +140,12 @@ public function getSearchByIdAndOwner(
         UserEntityInterface|int|null $userOrId
     ): ?SearchEntityInterface {
         $userId = $userOrId instanceof UserEntityInterface ? $userOrId->getId() : $userOrId;
-        $callback = function ($select) use ($id, $sessionId, $userId) {
-            $nest = $select->where
-                ->equalTo('id', $id)
-                ->and
-                ->nest
-                ->equalTo('session_id', $sessionId);
-            if (!empty($userId)) {
-                $nest->or->equalTo('user_id', $userId);
-            }
-        };
-        return $this->getDbTable('search')->select($callback)->current();
+        $entityClass = $this->getEntityClass(SearchEntityInterface::class);
+        $dql = 'SELECT s FROM ' . $entityClass . ' s '
+            . 'WHERE s.id = :id AND s.sessionId = :sessionId AND s.user = :userId';
+        $query = $this->entityManager->createQuery($dql);
+        $query->setParameters(compact('id', 'sessionId', 'userId'));
```

Comment:
Similar situation here:

```suggestion
        $parameters = compact('id', 'sessionId');
        $dql = 'SELECT s FROM ' . $entityClass . ' s '
            . 'WHERE s.id = :id AND s.sessionId = :sessionId';
        if ($userId !== null) {
            $dql .= ' AND s.user = :userId';
            $parameters['userId'] = $userId;
        }
        $query = $this->entityManager->createQuery($dql);
        $query->setParameters($parameters);
```